### ISSUE TITLE
pyaudio stream is not cleared when a trial is abandoned

### ISSUE THREAD

**Issue by neuromusic** [gentnerlab/pyoperant#77]
currently a trial has the following logic:
1. queue up a wav file to pyaudio
2. wait for a peck to initiate a trial
3. if the session should end, then end the session
4. otherwise, play the file

if a session ends, the wav data is still loaded into the pyaudio device, so when a new trial starts, the device is unavailable.

non-exclusive options:
- catch the pyaudio error & try to close the stream
- close the stream at the end of a session
- close the stream any time the panel is reset (not a bad idea)

basically, at the end of a session we need to  do everything in `pa_interface.close()` except `pa.terminate()`. perhaps, we should add a `reset()` method that `close()` calls then use `reset` any time we need to clear the stream
